src/Translator.js

```diff
@@ -56,17 +56,25 @@ class Translator extends EventEmitter {
     if (typeof keys === 'number') keys = String(keys);
     if (typeof keys === 'string') keys = [keys];
 
-    // return key on CIMode
-    let lng = options.lng || this.language;
-    if (lng && lng.toLowerCase() === 'cimode') return keys[keys.length - 1];
```

Comment:
My tests for when `appendNamespaceToCIMode: true` failed. This appears to be returning everything including the namespace if it is passed directly in the key (it is called before `this.extractFromKey`. I am not exactly sure why because this is the behavior I get in my application.

Below I started using `key` from `this.exctractFromKey`. Let me know if that was a bad idea.